devito/ir/iet/visitors.py

```diff
@@ -743,6 +743,12 @@ def __init__(self, mapper=None, replacer=None):
         super(XSubs, self).__init__()
         self.replacer = replacer or (lambda i: uxreplace(i, mapper))
 
+    def visit_Conditional(self, o):
+        condition = self.replacer(o.condition)
+        then_body = self._visit(o.then_body)
```

Comment:
As said on slack, I was not thinking near here. Thanks.

Reply:
> I was not thinking near here.

it's symptomatic you have to keep studying how this works